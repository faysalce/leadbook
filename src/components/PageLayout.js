import React, { Component } from 'react';
import { Layout } from 'antd';
import { connect } from 'react-redux';
import { withRouter, Switch, Route } from 'react-router-dom';
import Login from "./userAuth/Login"
import Contacts from "./Contacts";
import ErrorBoundary from "./ErrorBoundary";
const { Footer } = Layout;



class PageLayout extends Component {

  componentDidMount() {

    this.props.auth ? this.props.history.push("/contacts/all") : this.props.history.push("/login")


  }

  render() {

    return (
      <Layout className="layout">

        <ErrorBoundary>
          <Switch>
            <Route exact path='/login' component={() => <Login />} />
            <Route path='/contacts' component={() => <Contacts />} />

          </Switch>
        </ErrorBoundary>

        <Footer style={{ textAlign: 'center' }}>Lead Book Â©2019 Created Faysal</Footer>
      </Layout>

    );
  }
}


function mapStateToProps(store) {
  return {
    auth: store.user.auth,

  };
}
const mapDispatchToState = (dispatch, ownProps) => {
  return {
    // userLogin: (user,password) => dispatch(allContacts(user,password)),


  };
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToState
  )(PageLayout)
);