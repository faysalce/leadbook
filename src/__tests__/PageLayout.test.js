import React from 'react';
import PageLayout from '../components/PageLayout';
import {shallow} from "enzyme";

it("PageLayout renders correctly", () => {
  const wrapper = shallow(
    <PageLayout />
  );
  expect(wrapper).toMatchSnapshot();
});