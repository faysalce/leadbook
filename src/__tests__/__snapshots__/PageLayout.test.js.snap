// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders correctly 1`] = `ShallowWrapper {}`;
